src/common/snippets/src/generator.cpp

```diff
@@ -67,9 +67,7 @@ Generator::opRegType Generator::get_op_reg_type(const std::shared_ptr<Node>& op)
         std::dynamic_pointer_cast<op::LoopEnd>(op) ||
         std::dynamic_pointer_cast<op::Brgemm>(op) ||
         std::dynamic_pointer_cast<op::Buffer>(op) ||
-        std::dynamic_pointer_cast<op::RankNormalization>(op) ||
-        std::dynamic_pointer_cast<op::PerfCountBeginBase>(op) ||
-        std::dynamic_pointer_cast<op::PerfCountEndBase>(op))
+        std::dynamic_pointer_cast<op::RankNormalization>(op))
```

Comment:
We still need to assign a proper `reg_type` for PerfCount emitters, right? Then why `PerfCount*Base` were removed from this `if`?

Reply:
@IvanNovoselov , As you know, in essential there is no need to pass data with a register from perfCountBegin to perfCountEnd in runtime. perfCountEnd node has a class member perfCountBegin node, so can access it and interact directly. So this is the reason as discussed, we do not need port connector between perfCountBegin and perfCountEnd in the [insert example](https://github.com/openvinotoolkit/openvino/blob/master/src/common/snippets/src/lowered/pass/insert_perf_count.cpp#L53).
After re-thinking, some body may set port connector in their custom insertion for visualization or sth. In that case, we still need a dummy reg_type to work successfully, so I added it in [9ec2e9f](https://github.com/openvinotoolkit/openvino/pull/21494/commits/9ec2e9feaa1edb4cafca3c78cf14ebd4c4af521b). do you think it's right?

Reply:
@chenhu-wang, every operation present in LIR must be assigned a valid `reg_type`, it doesn't matter whether it has output connectors or not. 
Look, [here](https://github.com/openvinotoolkit/openvino/blob/261cf4e4dc653f935050902b2d1f66b403ce6ba5/src/common/snippets/src/lowered/pass/assign_registers.cpp#L32) `AssignRegisters` calls reg_type_mapper for every operation. The mapper is just wrapped `Generator::get_op_reg_type()` as you can see [here](https://github.com/openvinotoolkit/openvino/blob/de052d7d1095e650ae47657b4dccec28c89b0bc5/src/common/snippets/src/generator.cpp#L27C1-L27C36). Then, if an operation is not assigned a `reg_type` in the generator.cpp, it will fall through to `CPUGenerator::get_specific_op_reg_type()`. If it's not mapped even there, then an [exception](https://github.com/openvinotoolkit/openvino/blob/de052d7d1095e650ae47657b4dccec28c89b0bc5/src/plugins/intel_cpu/src/emitters/x64/cpu_generator.cpp#L238) will be thrown from `CPUGenerator::get_specific_op_reg_type()`.

Reply:
Thanks @IvanNovoselov for the explanation, reg_type have been re-added for the new ops.